### PR TITLE
Separate Jad challenges from Inferno

### PR DESCRIPTION
**Pull request by pajlada**
Relies on https://oldschool.runescape.wiki/w/RuneScape:Varplayer/2926 to check whether the player is inside the Jad challenge room since the region is the same

I have tested this in-game and it works as expected.
I have not changed any defaults on purpose, as I think deaths in this challenge may be a bit too frequent to warrant death notifications by default.

Changelog entry change is up for debate, as is naming

### CODE REVIEW

File: src/main/java/dinkplugin/domain/ExceptionalDeath.java [hunk 1]
```diff
@@ -10,7 +10,8 @@ public enum ExceptionalDeath {
     COX("Chambers of Xeric"),
     FIGHT_CAVE("Fight Caves"),
     INFERNO("Inferno"),
-    TOA("Tombs of Amascut");
+    TOA("Tombs of Amascut"),
+    JAD_CHALLENGES("Jad challenges");
```
Comment (iProdigy) [inline src/main/java/dinkplugin/domain/ExceptionalDeath.java:14]:
```suggestion
    JAD_CHALLENGES("Jad challenges"),
    TOA("Tombs of Amascut");
```

alphabetic order

File: src/main/java/dinkplugin/util/WorldUtils.java [hunk 2]
```diff
@@ -101,8 +107,16 @@ public boolean isGauntlet(int regionId) {
...
+    public boolean isInferno(Client client, int regionId) {
+        return regionId == INFERNO_REGION && client.getVarpValue(INSIDE_RAID_OR_CHALLENGE) == 0;
+    }
+
+    /**
+     * Checks whether the player is within TzHaar-Ket-Rak's Challenges
+     * The in-game region is the same as the inferno, but a varp value is different
+     */
+    public boolean isJadChallenges(Client client, int regionId) {
+        return regionId == INFERNO_REGION && client.getVarpValue(INSIDE_RAID_OR_CHALLENGE) == 1001;
```
Comment (iProdigy) [inline src/main/java/dinkplugin/util/WorldUtils.java:119]:
```suggestion
        return regionId == INFERNO_REGION && client.getVarpValue(INSIDE_RAID_OR_CHALLENGE) > 0;
```

to be safe

**Review by iProdigy — COMMENTED** [review]
good investigating

**Review by iProdigy — APPROVED** [review]
Code looks good to me!

File: src/main/java/dinkplugin/domain/ExceptionalDeath.java [hunk 3]
```diff
@@ -10,6 +10,7 @@ public enum ExceptionalDeath {
     COX("Chambers of Xeric"),
     FIGHT_CAVE("Fight Caves"),
     INFERNO("Inferno"),
+    JAD_CHALLENGES("Jad challenges"),
```
Comment (Felanbird) [inline src/main/java/dinkplugin/domain/ExceptionalDeath.java:13]:
```suggestion
    JAD_CHALLENGES("TzHaar-Ket-Rak Challenges"),
```
I'm fine keeping the enum name, but the user-facing one needs to be recognizable, Jad doesn't offer these challenges

Comment (Felanbird) [reply src/main/java/dinkplugin/domain/ExceptionalDeath.java:13]:
```suggestion
    JAD_CHALLENGES("Inferno Challenges"),
```
Discussed in chat, full nametoo long for the section of the settings. only other idea which seems more-correct

**Review by Felanbird — APPROVED** [review]
tested both and they both work 👍  - do whatever with the name, inferno is probably too long too

File: src/main/java/dinkplugin/domain/ExceptionalDeath.java (hunk 3 above)
Comment (pajlada) [reply src/main/java/dinkplugin/domain/ExceptionalDeath.java:13]: As discussed in chat, keeping as Jad challenges - can be changed later if people are super confused

**Pull request merged**